graviton/blackbox.py

```diff
@@ -269,32 +288,59 @@ def _to_bytes(cls, x, only_ints=False):
         return x.to_bytes(8, "big") if is_int else bytes(x, "utf-8")
 
     @classmethod
-    def _encode_arg(cls, arg, idx):
-        cls._assert_encodable(arg, f"problem encoding arg ({arg}) at index ({idx})")
+    def _encode_arg(cls, arg, idx, abi_type: Optional[abi.ABIType]) -> Optional[bytes]:
+        partial = cls._partial_encode_assert(
+            arg, abi_type, f"problem encoding arg ({arg}) at index ({idx})"
+        )
+        if partial is not None:
+            return partial
         return cls._to_bytes(arg, only_ints=True)
 
     @classmethod
-    def _assert_encodable(cls, arg: Any, msg: str = "") -> None:
+    def _partial_encode_assert(
+        cls, arg: Any, abi_type: Optional[abi.ABIType], msg: str = ""
+    ) -> Optional[bytes]:
+        if abi_type:
+            try:
+                return abi_type.encode(arg)
+            except Exception as e:
+                raise AssertionError(
+                    f"{msg +': ' if msg else ''}can't handle arg [{arg}] of type {type(arg)} and abi-type {abi_type}: {e}"
+                )
         assert isinstance(
-            arg, (int, str)
+            arg, (bytes, int, str)
```

Comment:
@barnjamin  - This is in the spirit of your PR  and allows `bytes` through unimpeded